cloudimg/aws.py

```diff
@@ -643,3 +703,92 @@ def tag_image(self, image, tags):
         res = image.create_tags(**attrs)
         log.debug("Tag image \"%s\" results: %s" % image.name, res)
         return res
+
+    def deregister_image(self, image):
+        """
+        Deregisters AMI image from AWS.
+
+        Args:
+            image (Image): An EC2 Image
+
+        Returns:
+            Id of deregistered image. (str)
+        """
+        image_id, image_name = image.id, image.name
+        logging.info("Deregistering image %s (%s)", image_id, image_name)
+
+        res = image.deregister()
```

Comment:
Is there some doubt about what it returns?  I checked the boto docs and it seems like it's always supposed to return None (i.e. it either succeeds, or raises an exception).

What I dislike about this code as it is now is that, at a glance, it looks buggy.  It calls a method and keeps/logs the return value, implying the value is expected to be somehow significant and not always the same thing.  But the value is not checked in any way, which raises the possibility that this is a case of doing something and then failing to check whether it succeeded.